torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -745,6 +870,61 @@ void propagateQuantizationOps(Block* block) {
   }
 }
 
+void InsertQuantDeQuantHelper::runWeightObserver(
+    Module& module,
+    const std::string& method_name) {
+  for (auto& invoked_methods : getInvokedMethods(module, method_name)) {
+    auto& invoked_module = std::get<0>(invoked_methods);
+    const auto& invoked_method_name = std::get<1>(invoked_methods);
+    runWeightObserver(invoked_module, invoked_method_name);
+  }
+  Method method = module.get_method(method_name);
+  auto graph = method.graph();
+  Value* self = graph->inputs()[0];
+
+  std::vector<Value*> weight_values;
+  for (size_t idx = 1; idx < method.num_inputs(); ++idx) {
+    auto& v = graph->inputs()[idx];
+    if (v->type()->isSubtypeOf(TensorType::get())) {
+      auto observer_name = findObserverName(v);
+      // Ensure that we only track values that are the output of observer node.
+      // isWeight returns true if the observed value is a weight of quantized
+      // op.
+      if (observer_name && isWeight(module, v)) {
```

Comment:
wait, how would input of the graph the output of the observer module?

Reply:
Can remove this